firedrake/solving.py

```diff
@@ -215,6 +203,13 @@ def _la_solve(A, x, b, **kwargs):
 
     bcs, solver_parameters, nullspace, nullspace_T, near_nullspace, \
         options_prefix = _extract_linear_solver_args(A, x, b, **kwargs)
+
+    from firedrake.bcs import FormBC
+    if bcs is not None:
+        for bc in bcs:
+            if isinstance(bc, FormBC) and not (isinstance(bc.formeq.lhs, ufl.Form) and isinstance(bc.formeq.rhs, ufl.Form)):
```

Comment:
Prefer to check this in the `FormBC` constructor. Then you have to sprinkle fewer `isinstance` checks here. You "know" the value is good.